<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GUIPlayerClass</class>
 <widget class="QMainWindow" name="GUIPlayerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>438</width>
    <height>301</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Drumstick MIDI File Player Multiplatform</string>
  </property>
  <property name="windowIcon">
   <iconset resource="guiplayer.qrc">
    <normaloff>:/dmidiplayer.png</normaloff>:/dmidiplayer.png</iconset>
  </property>
  <widget class="QWidget" name="MainWidget">
   <property name="acceptDrops">
    <bool>true</bool>
   </property>
   <property name="windowTitle">
    <string>Drumstick MIDI Player</string>
   </property>
   <property name="windowIcon">
    <iconset>
     <normaloff>:/resources/drumstick.png</normaloff>:/resources/drumstick.png</iconset>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0" rowspan="2" colspan="2">
     <widget class="QFrame" name="frame">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="palette">
       <palette>
        <active>
         <colorrole role="WindowText">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Text">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Base">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Window">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="PlaceholderText">
          <brush brushstyle="NoBrush">
           <color alpha="128">
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
        </active>
        <inactive>
         <colorrole role="WindowText">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Text">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Base">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Window">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="PlaceholderText">
          <brush brushstyle="NoBrush">
           <color alpha="128">
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
        </inactive>
        <disabled>
         <colorrole role="WindowText">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>146</red>
            <green>145</green>
            <blue>144</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Text">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>165</red>
            <green>164</green>
            <blue>164</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Base">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="Window">
          <brush brushstyle="SolidPattern">
           <color alpha="255">
            <red>0</red>
            <green>0</green>
            <blue>0</blue>
           </color>
          </brush>
         </colorrole>
         <colorrole role="PlaceholderText">
          <brush brushstyle="NoBrush">
           <color alpha="128">
            <red>255</red>
            <green>255</green>
            <blue>255</blue>
           </color>
          </brush>
         </colorrole>
        </disabled>
       </palette>
      </property>
      <property name="statusTip">
       <string>Playback time and current levels</string>
      </property>
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="spacing">
        <number>4</number>
       </property>
       <item row="0" column="0" rowspan="3">
        <widget class="QLabel" name="lblTime">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>210</width>
           <height>62</height>
          </size>
         </property>
         <property name="baseSize">
          <size>
           <width>210</width>
           <height>62</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>Sans Serif</family>
           <pointsize>36</pointsize>
           <kerning>true</kerning>
          </font>
         </property>
         <property name="text">
          <string>00:00:00</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="lbl3">
         <property name="text">
          <string>Tempo:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="lblOther">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>100</width>
           <height>0</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="lbl4">
         <property name="text">
          <string>Volume:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="lblVolume">
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="lbl5">
         <property name="text">
          <string>Pitch:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSpinBox" name="spinPitch">
         <property name="maximumSize">
          <size>
           <width>50</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="toolTip">
          <string>Pitch Control</string>
         </property>
         <property name="statusTip">
          <string>Pitch transpose between -12 and +12 semitones</string>
         </property>
         <property name="styleSheet">
          <string notr="true">color: white;
background-color: black;
selection-color: white;
selection-background-color: blue;</string>
         </property>
         <property name="minimum">
          <number>-12</number>
         </property>
         <property name="maximum">
          <number>12</number>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="btnVolume">
      <property name="toolTip">
       <string>Reset Volume</string>
      </property>
      <property name="statusTip">
       <string>Reset Volume to 100%</string>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="icon">
       <iconset resource="guiplayer.qrc">
        <normaloff>:/resources/vol.png</normaloff>:/resources/vol.png</iconset>
      </property>
     </widget>
    </item>
    <item row="1" column="2" rowspan="4">
     <widget class="QSlider" name="volumeSlider">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="statusTip">
       <string>Volume Slider</string>
      </property>
      <property name="maximum">
       <number>200</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="tracking">
       <bool>false</bool>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksAbove</enum>
      </property>
     </widget>
    </item>
    <item row="2" column="0" colspan="2">
     <widget class="QProgressBar" name="progressBar">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="statusTip">
       <string>Playback progress</string>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="textVisible">
       <bool>true</bool>
      </property>
      <property name="textDirection">
       <enum>QProgressBar::TopToBottom</enum>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QSlider" name="sliderTempo">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="statusTip">
       <string>Tempo Slider</string>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>200</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="pageStep">
       <number>10</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="tickPosition">
       <enum>QSlider::TicksAbove</enum>
      </property>
      <property name="tickInterval">
       <number>5</number>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QPushButton" name="btnTempo">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>Reset Tempo</string>
      </property>
      <property name="statusTip">
       <string>Reset Tempo to 100%</string>
      </property>
      <property name="text">
       <string>tempo=100%</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::ExpandingFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="lbl1">
        <property name="text">
         <string>File Name:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="lblName">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="statusTip">
         <string>Currently loaded MIDI file name</string>
        </property>
        <property name="openExternalLinks">
         <bool>true</bool>
        </property>
        <property name="textInteractionFlags">
         <set>Qt::LinksAccessibleByKeyboard|Qt::LinksAccessibleByMouse|Qt::TextBrowserInteraction|Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="3" column="1">
     <widget class="RhythmView" name="rhythm" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>438</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuRecentFiles">
     <property name="title">
      <string>Open Recent Files</string>
     </property>
    </widget>
    <addaction name="actionOpen"/>
    <addaction name="actionFileInfo"/>
    <addaction name="menuRecentFiles"/>
    <addaction name="separator"/>
    <addaction name="actionPlay"/>
    <addaction name="actionPause"/>
    <addaction name="actionStop"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <widget class="QMenu" name="menuLanguage">
     <property name="title">
      <string>&amp;Language</string>
     </property>
    </widget>
    <addaction name="menuLanguage"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
    <addaction name="actionAboutQt"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionShowToolbar"/>
    <addaction name="actionShowStatusbar"/>
    <addaction name="separator"/>
    <addaction name="actionMIDISetup"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionPlaylist"/>
    <addaction name="actionLyrics"/>
    <addaction name="actionRhythm"/>
    <addaction name="actionChannels"/>
    <addaction name="actionPianoPlayer"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSettings"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>Tool Bar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>BottomToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionMIDISetup"/>
   <addaction name="separator"/>
   <addaction name="actionPlay"/>
   <addaction name="actionPause"/>
   <addaction name="actionStop"/>
   <addaction name="separator"/>
   <addaction name="actionAbout"/>
   <addaction name="actionQuit"/>
  </widget>
  <widget class="QStatusBar" name="myStatusBar"/>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="guiplayer.qrc">
     <normaloff>:/resources/open.png</normaloff>:/resources/open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="statusTip">
    <string>Open a MIDI file</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="guiplayer.qrc">
     <normaloff>:/resources/quit.png</normaloff>:/resources/quit.png</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="statusTip">
    <string>Quit the application</string>
   </property>
  </action>
  <action name="actionPlay">
   <property name="icon">
    <iconset resource="guiplayer.qrc">
     <normaloff>:/resources/play.png</normaloff>:/resources/play.png</iconset>
   </property>
   <property name="text">
    <string>Play</string>
   </property>
   <property name="statusTip">
    <string>Start playing the current MIDI file</string>
   </property>
   <property name="autoRepeat">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionPause">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="guiplayer.qrc">
     <normaloff>:/resources/pause.png</normaloff>:/resources/pause.png</iconset>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="statusTip">
    <string>Pause the playback</string>
   </property>
   <property name="autoRepeat">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionStop">
   <property name="icon">
    <iconset resource="guiplayer.qrc">
     <normaloff>:/resources/stop.png</normaloff>:/resources/stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="statusTip">
    <string>Stop the playback</string>
   </property>
   <property name="autoRepeat">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="guiplayer.qrc">
     <normaloff>:/resources/about.png</normaloff>:/resources/about.png</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
   <property name="statusTip">
    <string>Show the about box</string>
   </property>
  </action>
  <action name="actionAboutQt">
   <property name="icon">
    <iconset resource="guiplayer.qrc">
     <normaloff>:/resources/qtlogo.png</normaloff>:/resources/qtlogo.png</iconset>
   </property>
   <property name="text">
    <string>About Qt</string>
   </property>
   <property name="statusTip">
    <string>Show the about Qt dialog box</string>
   </property>
  </action>
  <action name="actionMIDISetup">
   <property name="icon">
    <iconset resource="guiplayer.qrc">
     <normaloff>:/resources/setup.png</normaloff>:/resources/setup.png</iconset>
   </property>
   <property name="text">
    <string>MIDI Setup</string>
   </property>
   <property name="statusTip">
    <string>Select a connection for the MIDI output port</string>
   </property>
  </action>
  <action name="actionShowToolbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Tool Bar</string>
   </property>
   <property name="toolTip">
    <string>Show Tool Bar</string>
   </property>
   <property name="statusTip">
    <string>Show or hide the tool bar</string>
   </property>
  </action>
  <action name="actionShowStatusbar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Status Bar</string>
   </property>
   <property name="statusTip">
    <string>Show or hide the status bar</string>
   </property>
  </action>
  <action name="actionLyrics">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Lyrics and Texts</string>
   </property>
   <property name="toolTip">
    <string>Show or Hide Lyrics and Texts</string>
   </property>
   <property name="statusTip">
    <string>Show or Hide Lyrics and Texts</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionPlaylist">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Playlist</string>
   </property>
   <property name="toolTip">
    <string>Show or Hide Playlist</string>
   </property>
   <property name="statusTip">
    <string>Show or Hide Playlist</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionChannels">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Channels</string>
   </property>
   <property name="toolTip">
    <string>Show or Hide Channels</string>
   </property>
   <property name="statusTip">
    <string>Show or Hide Channels</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionPianoPlayer">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Piano Player</string>
   </property>
   <property name="toolTip">
    <string>Show or Hide Piano Player</string>
   </property>
   <property name="statusTip">
    <string>Show or Hide Piano Player</string>
   </property>
  </action>
  <action name="actionRhythm">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Rhythm</string>
   </property>
   <property name="toolTip">
    <string>Show or Hide Rhythm</string>
   </property>
   <property name="statusTip">
    <string>Show or Hide Rhythm</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionFileInfo">
   <property name="text">
    <string>File Info</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RhythmView</class>
   <extends>QWidget</extends>
   <header>rhythmview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>spinPitch</tabstop>
  <tabstop>sliderTempo</tabstop>
  <tabstop>btnTempo</tabstop>
  <tabstop>volumeSlider</tabstop>
 </tabstops>
 <resources>
  <include location="guiplayer.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionShowStatusbar</sender>
   <signal>toggled(bool)</signal>
   <receiver>myStatusBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShowToolbar</sender>
   <signal>toggled(bool)</signal>
   <receiver>toolBar</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>51</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
